sda-commons-server-morphia/src/main/java/org/sdase/commons/server/morphia/MorphiaBundle.java

```diff
@@ -54,23 +57,30 @@
   private final boolean ensureIndexes;
   private final boolean forceEnsureIndexes;
   private final Tracer tracer;
-
-  /** Activate JSR303 validation. */
+  /**
+   * Activate JSR303 validation.
+   */
   private final boolean activateValidation;
 
+  private final CaCertificatesBundle.FinalBuilder<C> caCertificatesBundleBuilder;
+  private final boolean useCaCertificateBundle;
   private MongoClient mongoClient;
   private Datastore morphiaDatastore;
+  private SSLContext sslContext;
+  private CaCertificatesBundle<C> caCertificatesBundle;
 
   private MorphiaBundle( // NOSONAR: Methods should not have too many parameters
-      MongoConfigurationProvider<C> configProvider,
-      Set<String> packagesToScan,
-      @SuppressWarnings("rawtypes") // same type as in Morphia
-          Set<Class> entityClasses,
-      Set<TypeConverter> customConverters,
-      boolean ensureIndexes,
-      boolean forceEnsureIndexes,
-      boolean activateValidation,
-      Tracer tracer) {
+                         MongoConfigurationProvider<C> configProvider,
+                         Set<String> packagesToScan,
+                         @SuppressWarnings("rawtypes") // same type as in Morphia
+                             Set<Class> entityClasses,
+                         Set<TypeConverter> customConverters,
+                         boolean ensureIndexes,
+                         boolean forceEnsureIndexes,
+                         boolean activateValidation,
+                         Tracer tracer,
+                         CaCertificatesBundle.FinalBuilder<C> caCertificatesBundleBuilder,
+                         boolean useCaCertificateBundle) {
```

Comment:
I think, if we prefer the existing certificates as preferred, we can remove this boolean advise. We can then always provide the CaCertificatesBundle with default config and refactor `withCaCertificateConfigProvider` by `withCaCertificatesFromBundle(CaCertificateBundle<C> caCertificateBundle)` to use a globally configured one.

However, I like the idea to use the new bundle here directly so that it is not necessary for the consumer to create one on his own. But how we do that exactly is probably a topic we should talk about.